DataDog/dd-trace-rb
# [NO-TICKET] Reduce startup logs verbosity

**What does this PR do?**

With the change in #3441 to print the library configuration on every configuration, we were concerned that it may generate noisy logs.

This PR improves the situation by:
* Tweaking the date format (using a few less characters)
* Introducing a `extra_fields` argument in the `EnvironmentLogger`, and using it to avoid the need for a separate profiler logger entry.

Thus changing this:

```
$ bundle exec ruby -e "require 'ddtrace'; 2.times { Datadog.configure { } }"
I, [2024-02-08T10:29:37.145777 #394977]  INFO -- ddtrace: [ddtrace] DATADOG CONFIGURATION - PROFILING - {"profiling_enabled":false}
I, [2024-02-08T10:29:37.150765 #394977]  INFO -- ddtrace: [ddtrace] DATADOG CONFIGURATION - CORE - {"date":"2024-02-08T10:29:37+00:00","os_name":"x86_64-pc-linux-gnu","version":"1.20.0","lang":"ruby","lang_version":"3.1.4","env":null,"service":"-e","dd_version":null,"debug":false,"tags":null,"runtime_metrics_enabled":false,"vm":"ruby-3.1.4","health_metrics_enabled":false}
I, [2024-02-08T10:29:37.152910 #394977]  INFO -- ddtrace: [ddtrace] DATADOG CONFIGURATION - PROFILING - {"profiling_enabled":false}
I, [2024-02-08T10:29:37.153244 #394977]  INFO -- ddtrace: [ddtrace] DATADOG CONFIGURATION - CORE - {"date":"2024-02-08T10:29:37+00:00","os_name":"x86_64-pc-linux-gnu","version":"1.20.0","lang":"ruby","lang_version":"3.1.4","env":null,"service":"-e","dd_version":null,"debug":false,"tags":null,"runtime_metrics_enabled":false,"vm":"ruby-3.1.4","health_metrics_enabled":false}
```

into

```
I, [2024-02-08T10:29:19.137025 #394537]  INFO -- ddtrace: [ddtrace] DATADOG CONFIGURATION - CORE - {"date":"2024-02-08T10:29:19Z","os_name":"x86_64-pc-linux-gnu","version":"1.20.0","lang":"ruby","lang_version":"3.1.4","env":null,"service":"-e","dd_version":null,"debug":false,"tags":null,"runtime_metrics_enabled":false,"vm":"ruby-3.1.4","health_metrics_enabled":false,"profiling_enabled":false}
I, [2024-02-08T10:29:19.139572 #394537]  INFO -- ddtrace: [ddtrace] DATADOG CONFIGURATION - CORE - {"date":"2024-02-08T10:29:19Z","os_name":"x86_64-pc-linux-gnu","version":"1.20.0","lang":"ruby","lang_version":"3.1.4","env":null,"service":"-e","dd_version":null,"debug":false,"tags":null,"runtime_metrics_enabled":false,"vm":"ruby-3.1.4","health_metrics_enabled":false,"profiling_enabled":false}
```

**Motivation:**

Startup logs are useful, but should be as minimal as possible.

**Additional Notes:**

I've introduced this change in a generic way -- e.g. the core only needs to know that components can return extra fields, and fold them into the things that get reported by the environment logger.

**How to test the change?**

Change includes test coverage; you can also use the example above:

> `bundle exec ruby -e "require 'ddtrace'; 2.times { Datadog.configure { } }"`

to see it running locally.

**For Datadog employees:**

credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
